Generator/Sources/generate/Unfolder.swift

```diff
@@ -0,0 +1,72 @@
+import SDGControlFlow
+import SDGText
+import SDGHTML
+import SDGWeb
+
+struct Unfolder: SiteSyntaxUnfolder {
+
+  // MARK: - Properties
+
+  private let standardUnfolder: SyntaxUnfolder
+
+  private static let frame: ListSyntax<ContentSyntax> = {
+    let file = repositoryStructure.components.appendingPathComponent("Frame.html")
+    let document: DocumentSyntax
+    do {
+      let source = try StrictString(from: file)
+      document = try DocumentSyntax.parse(source: String(source)).get()
+    } catch {
+      fatalError("\(error)")
+    }
+    return document.content
+  }()
+  private let content: Shared<ListSyntax<ContentSyntax>?> = Shared(nil)
+
+  // MARK: - SyntaxUnfolderProtocol
+
+  init(context: SyntaxUnfolder.Context) {
+    standardUnfolder = SyntaxUnfolder(context: context)
+  }
+
+  func unfold(element: inout ElementSyntax) throws {
+
+    if let content = self.content.value,
+      element.nameText == "content" {
+      element.nameText = "div"
+      element.content = content
+    }
+
+    if element.nameText == "laccPage" {
+      self.content.value = element.content
+      defer { self.content.value = nil }
+      element.nameText = "page"
+      element.content = Unfolder.frame
+      // Conduct another pass now that we know the content.
+      try element.unfold(with: self)
+    }
+
+    if element.nameText == "sermon" {
```

Comment:
HTML is handled as a syntax tree now instead of text. This accidental syntax errors less likely, and provides a built‐in way of parsing your own custom information. Instead of find‐and‐replace, the syntax tree is fed into a “syntax unfolder”, which replaces entire nodes in the tree according to their content.

For example, this is the new implementation of your sermon rows. Every `<sermon>` pseudo‐element will be replaced with the contents of your template, with the template making use of the `date`, `title` and `passage` pseudo‐attributes.